### PR TITLE
Adding Tools - Captioning, Image Processing, Generic OD & Seg

### PR DESCRIPTION
**Pull request by shankar-vision-eng**
This PR includes the following features
- Adds capability to `extract_frames` to extract frames from youtube videos
- Adds Florancev2 image captioning in two modes - Brief and Detailed.
- Adds Generic OD and Segmentation tool to detect common objects with text prompts
- Adds Florancev2 based image question answering with context
- Adds following image generation/processing tools
- - Image to Depth - Use Depth Anything V2
- - Image to Pose - Generates Open pose stick image which can interpreted by Open pose libs
- - Image to Soft Edge Image - Generates HED or Soft edge image of object boundaries 
- - Image to Normal - Generates Normal image where RGB values represent XYZ surface normals
- Adds template matching tool to find cropped ROI in original image

### CODE REVIEW

File: vision_agent/tools/tools.py [hunk 1]
```diff
@@ -521,6 +544,309 @@ def blip_image_caption(image: np.ndarray) -> str:
...
+    Returns:
+        List[Dict[str, Any]]: A list of dictionaries containing the score and
+            bounding box of the detected template with normalized coordinates between 0
+            and 1 (xmin, ymin, xmax, ymax). xmin and ymin are the coordinates of the
+            top-left and xmax and ymax are the coordinates of the bottom-right of the
+            bounding box.
+
+    Example
+    -------
+        >>> >>> template_match(image, template)
```
Comment (dillonalaird) [inline vision_agent/tools/tools.py:823]: NIT extra `>>>` you can also check the documentation by running `poetry run mkdocs build -f mkdocs.yml -d docs-build`

Comment (shankar-vision-eng) [reply vision_agent/tools/tools.py:823]: Oh nice!! I did not know. Will do

**Review by dillonalaird — CHANGES_REQUESTED** [review]
PR looks good, but can you use the model names in to tool function names? For example `image_question_answering_with_context` and the generic tools

**Comment by shankar-vision-eng** [comment]
> PR looks good, but can you use the model names in to tool function names? For example `image_question_answering_with_context` and the generic tools

Renamed tools with actual model names

**Review by dillonalaird — APPROVED** [review]
LGTM

**Pull request merged**